git/branch_manager.go

```diff
@@ -0,0 +1,96 @@
+package git
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/drewwyatt/git-tidy/reporting"
+)
+
+type branchDeletionError struct {
+	Branch string
+	Msg    string
+}
+
+var reporter = reporting.NewReporter()
+
+// BranchManager higher level abstraction of Git
+type BranchManager struct {
+	git Git
+
+	GoneBranches         []string
+	DeletedBranches      []string
+	BranchDeletionErrors []branchDeletionError
+}
+
+// NewBranchManager Constructor for branch manager
+func NewBranchManager(directory string, exec Executor) BranchManager {
+	git := Git{directory: directory, exec: exec}
+	return BranchManager{git: git}
+}
+
+// Init run setup git commands and set gone branches
+func (bm *BranchManager) Init() []error {
```

Comment:
I wouldn't say I _love_ the name of this function because it sounds like it would be a conventional OOP initializer/constructor, but it's not. Might be that other languages are coloring me, though.

Reply:
I think this makes sense - but I can't come up with anything that doesn't sound exactly the same (e.g. `setup`).

I could just move all of this into `NewBranchManager`?

Reply:
"Setup" or "configure" doesn't sound exactly the same to me. All I really mean is that "init" is a reserved keyword in other languages.